packages/circuits-o1js/test/unit/o1jsZkPrograms/Reserves-Circuit.test.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  ForeignCurve,
+  CanonicalForeignField,
+  MerkleTree,
+  Poseidon,
+  Field,
+  Encoding,
+} from "o1js";
+
+import {
+  Secp256k1,
+  Ecdsa,
+  Reserves,
+  Bytes32,
+} from "@circuits-o1js/Reserves.js";
+
+// helper function build generating a bunch of key pairs
+class PublicPrivateKeyMocks {
+  static generateKeys(count: number): [ForeignCurve, CanonicalForeignField][] {
+    const keypairs = new Array<[ForeignCurve, CanonicalForeignField]>();
+
+    for (let i = 0; i < count; i++) {
+      const privateKey = Secp256k1.Scalar.random();
+      const publicKey = Secp256k1.generator.scale(privateKey);
```

Comment:
While running these tests, I encountered a TypeError: os.availableParallelism is not a function error. It might be an issue on my end.

Reply:
I didn't run into that. Thanks for running the tests though. Let me try again on my side.